### PR TITLE
New Feature - Unicode property escape

### PR DESCRIPTION
**Pull request by kisbg**
Signed-off-by: bence gabor kis <kisbg@inf.u-szeged.hu>

### CODE REVIEW

File: third_party/yarr/UnicodePatternTables.h [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
-* Copyright (C) 2017 Apple Inc. All rights reserved.
+* Copyright (C) 2017-2019 Apple Inc. All rights reserved.
```
Comment (clover2123) [inline third_party/yarr/UnicodePatternTables.h]: Why did you add `2019` at here?

File: tools/test/test262/excludelist.orig.xml [hunk 2]
```diff
@@ -1488,402 +1488,11 @@
...
-    <test id="built-ins/RegExp/property-escapes/generated/Sentence_Terminal"><reason>TODO</reason></test>
-    <test id="built-ins/RegExp/property-escapes/generated/Soft_Dotted"><reason>TODO</reason></test>
-    <test id="built-ins/RegExp/property-escapes/generated/Terminal_Punctuation"><reason>TODO</reason></test>
-    <test id="built-ins/RegExp/property-escapes/generated/Unified_Ideograph"><reason>TODO</reason></test>
-    <test id="built-ins/RegExp/property-escapes/generated/Uppercase"><reason>TODO</reason></test>
-    <test id="built-ins/RegExp/property-escapes/generated/Variation_Selector"><reason>TODO</reason></test>
-    <test id="built-ins/RegExp/property-escapes/generated/White_Space"><reason>TODO</reason></test>
-    <test id="built-ins/RegExp/property-escapes/generated/XID_Continue"><reason>TODO</reason></test>
-    <test id="built-ins/RegExp/property-escapes/generated/XID_Start"><reason>TODO</reason></test>
+    <test id="built-ins/RegExp/property-escapes/generated/Assigned"><reason>Not Working on 12.0 unicode and 12.1 unicode table</reason></test>
```
Comment (ksh8281) [inline tools/test/test262/excludelist.orig.xml]: Does generating `UnicodePatternTables.h` with latest platform(with new version of ICU) fix this issue?

File: third_party/yarr/UnicodePatternTables.h (hunk 1 above)
Comment (kisbg) [reply third_party/yarr/UnicodePatternTables.h]: I generated the `UnicodePatternTables.h` with the `generateYarrUnicodePropertyTables.py` from webkit's repository. The script generated the new license header. 

https://github.com/WebKit/webkit/blob/master/Source/JavaScriptCore/yarr/generateYarrUnicodePropertyTables.py

**Comment by ksh8281** [comment]
@kisbg could you fix conflicts? if you favor, we can fix indention of third_party/yarr/YarrUnicodeProperties.cpp
